### ISSUE TITLE
Could it load and tune falcon-40B ?

### ISSUE THREAD

**Issue by znsoftm** [artidoro/qlora#120]
TIIUAC  have released a promising model named falcon: https://huggingface.co/tiiuae
Can QLora load and tune it?   It is totally free and can be used for commercial purposes.

**Comment by phalexo** [comment]
I loaded falcon-40b-instruct into 4 GPUs, 12.3GiB each. 

It runs extremely slowly.

I have seen others reporting loading it into a single GPU with 48GiB VRAM and it
was still extremely slow.

**Comment by znsoftm** [comment]
yes. I tested it with 8* A100, it occupates 12G on each GPU and stil not fast, especiaally alter the input length more than 500.

**Comment by phalexo** [comment]
> Yes. I tested it with 8* A100, it occupates 12G on each GPU and still not fast, especially alter the input length more than 500.

If falcon-40b produces high quality output, I would consider using it a source of training data for a smaller model.

**Comment by mirdulagarwal1201** [comment]
I dont know how to run a model using multiple GPU nodes for inference can someone give me the code for it ?
